get rid of jQuery usage

frostbolt opened the pull request:
Hi.

Your plugin depends on outdated jQuery version, that is no longer maintained
At first I was going to move it to third jQuery but then I noticed that there's only one jQuery call that can be easily replaced with native code.

As you can see [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map), native `.map` realization is compatible with all modern (IE9+) browsers.

atais commented:
dude the `eonasdan-datetimepicker` uses jQuery.
this is just a wrapper. 

you will need jQuery anyway to run the library.

frostbolt commented:
yeah, but it [uses](
https://github.com/Eonasdan/bootstrap-datetimepicker/blob/70eacba603023a511a611ed9cec277b379f655a5/package.json#L16) jquery itself.
your implementation requires an outdated 2nd version, which has security vulnerabilities, according to `npm audit`.

It's required to get rid of it's usage or update the dependency.

atais commented:
Sorry I did not have time to merge it
however, i do not know why but travis is not showing up here.
But your changes fail on tests:
https://travis-ci.org/atais/angular-eonasdan-datetimepicker

if you fix it I can merge the change.
I think fixing the dependency is easier

atais commented:
@frostbolt now travis works

frostbolt commented:
@atais I'll take a look later this month.

frostbolt commented:
Hi, @atais,
Actually I found an issue in my code 🌚 , but tests still doesn't work.

When I try to run them it seems like angularjs doesn't work. I can see `{{expressions}}` inside the page.

But the most strange thing is that I've managed to run them manually running `npm start`. Looks like everything is just fine.
![image](https://user-images.githubusercontent.com/6280096/56989432-19e43580-6b9b-11e9-80a7-dfedf713fa07.png)


PS. I've tried to run the tests for [origin/master](https://github.com/atais/angular-eonasdan-datetimepicker/commit/6daa877b5e6e132232e39ed58ae4cd3fa26330d3). Unsuccessfully.

atais commented:
Damn it happens from time to time because I don't update the dependencies and the tests have to be fixed. 
I am currently on holidays but I will try to fix it once I am back

frostbolt commented:
I noticed that I've accidentally moved several dependencies from dev to common section. Fixed that. 

I've also replaced `'` with `\"` as long as 1st variant causing error on windows platform (see screenshot below).
![image](https://user-images.githubusercontent.com/6280096/57307774-d420f300-70ed-11e9-9ccf-1fcc5d4078e2.png)

atais commented:
I have just looked into it, but it does not seem easy.
I do not know why it stopped working and what is the cause.

To be honest, I am not working with those technologies anymore so I really do not want to step into it.

If you want and could fix the tests I will gladly merge the changes.
